lms/djangoapps/branding/tests/test_views.py

```diff
@@ -315,3 +316,26 @@ def test_header_logo_links_to_marketing_site_with_site_override(self):
         self.client.login(username=self.user.username, password="password")
         response = self.client.get(reverse("dashboard"))
         self.assertIn(self.site_configuration_other.values["MKTG_URLS"]["ROOT"], response.content)
+
+    def test_index_redirects_to_signin_with_site_override(self):
+        """ Test index view redirects if ALWAYS_REDIRECT_HOMEPAGE_TO_SIGNIN_FOR_ANONYMOUS_USER is set in SiteConfiguration """
+        self.use_site(self.site_for_signin_redirection)
+        response = self.client.get(reverse("root"))
+        self.assertRedirects(
+            response,
+            reverse('signin_user'),
```

Comment:
shouldn't we check for `index_page` instead of the `signin_user` page?
Both unit tests are checking same page here.

Reply:
no, as we are trying to redirect to `signin_user` page to we have to check that